### ISSUE TITLE
[multi-top,sw,tools] Fix make_new_dif and related tools to not assume single top

### ISSUE THREAD

**Issue by a-will** [lowRISC/opentitan#25752]
### Description

make_new_dif and the other software test auto-generation tools currently assume the entire repo is dedicated to a single top. The output goes to a single place and the c preprocessor includes and bazel targets all assume a global IP namespace and directory structure. The reality is starkly different, with ipgen'd IPs and top-specific IPs in the individual top directories.

We'll need to go back and design how this all should work, and implement it. Afterwards, we can integrate Darjeeling into the flow.

**Comment by pamaury** [comment]
This is fixed in the multitop_dev branch and will be backported once #25580 is merged

**Comment by andreaskurth** [comment]
@pamaury: Is it correct that this is now done? Or are there open parts to this issue?

**Comment by pamaury** [comment]
I think that we can close this. There are particular DIFs which are problematic like `otp_ctrl` but the generic multitop mechanism is already merged. @a-will is it ok to close?

**Comment by a-will** [comment]
> I think that we can close this. There are particular DIFs which are problematic like `otp_ctrl` but the generic multitop mechanism is already merged. [@a-will](https://github.com/a-will) is it ok to close?

If we're going to split off handling ipgen'd IPs, we should probably create a separate issue and track it there. Do we have an answer for those problematic DIFs?

If we split off handling the problematic DIFs to another issue, we could close this one.

**Comment by pamaury** [comment]
For those DIFs (probably `otp_ctrl`), the idea would be to generate the DIF's code using a template. We could either create a new rule or (better) rely on #26555 which should make it easier to render templates that depend on top's information

**Comment by andreaskurth** [comment]
Thx both, then let's keep this open unless/until we decide to create a new issue to track the DIF for ipgen'd IPs.
